lib/server.js

```diff
@@ -15,9 +16,28 @@ function createServer (options, httpHandler) {
   return { server, listen }
 
   // `this` is the Fastify object
-  function listen () {
-    const listenOptions = normalizeListenArgs(Array.from(arguments))
-    const { cb, host } = listenOptions
+  function listen (listenOptions, ...args) {
+    let cb = args.slice(-1).pop()
+    // The `else if` and `else` branches go away when the variadic signature
```

Comment:
```suggestion
    // TO DO: The `else if` and `else` branches go away when the variadic signature
```

Reply:
Thank you, but I don't think we need `todo` notes when we know the feature is slated for removal.

Reply:
No strong opinion. I have noticed the feature will remove until `fastify@5`. So I'd prefer to mark as a `to do`. But I think It's okay to keep it as it is

Reply:
The deprecation _is_ the todo.